### PR TITLE
CLI - codegen command now supports the `--watch` option

### LINKED ISSUE #1459: Implement a polywrap codegen --watch  CLI command

### PR DESCRIPTION
**Pull request by pileks**
Resolves #1459

You can now run `npx polywrap codegen --watch` which will automatically watch files within your project directory and re-run codegen whenever changes are detected.

### CODE REVIEW

File: packages/cli/src/commands/codegen.ts
Comment (krisbitney) [inline packages/cli/src/commands/codegen.ts:150]: There is already a try-catch block within `codeGenerator.generate()` that catches and logs any errors, so the try-catch block in `codegen.ts` will log the errors a second time. 

https://github.com/polywrap/toolchain/blob/origin-dev/packages/cli/src/lib/codegen/CodeGenerator.ts

**Review by krisbitney — CHANGES_REQUESTED** [review]
This is a great feature! Very cool stuff. 😁

File: packages/cli/src/commands/codegen.ts
Comment (dOrgJelli) [reply packages/cli/src/commands/codegen.ts:150]: Hmmm, this try/catch logic is also in the build command. Honestly I think it's alright to log more than once, and it might be best to remove the try/catch from within the `lib` layer. Thoughts?

**Review by dOrgJelli — APPROVED** [review]
Looks great! Waiting for @krisbitney's approval regarding logging practices.

File: packages/cli/src/commands/codegen.ts
Comment (krisbitney) [reply packages/cli/src/commands/codegen.ts:150]: Yeah, either way is good. I prefer removing the try-catch from the lib layer if there are no side effects from doing that, but I'd also be happy merging this as-is.

**Review by krisbitney — APPROVED** [review]

**Pull request merged**
